### DISCUSSION TITLE
Is recompressing my repo feasible with borg 1.2?

### DISCUSSION THREAD

**Discussion by mbwgh** [borgbackup/borg#7848] [Q&A] [answered]
I am currently in the process of running `borg recreate /my/repo::$archive --recompress --compression zstd,9` for all of my archives, in order of the time they were created.

I am now ~50 archives in (of about 200) and things are getting ever slower with each new archive it seems. I am talking about 3-4 hours per archive.

This may be because more data is accumulated with successive backups of the same computer (i.e. later archives being larger).
But do chunk comparisons maybe become more expensive too, as the proportion of zstd-compressed archives increases?

If the latter was the case, maybe what I am trying to do is not actually feasible (having to wait several days per archive in the end).

I also saw that some support for repo-wide compression has been added to borg2. How long will it be until borg2 is officially released?

**Comment by ThomasWaldmann** [comment, marked as answer]
AFAIK, it should not get (significantly) slower (if you have enough RAM, so it doesn't do swapping).

Some fluctuations are possible though, e.g. the speed of the hash table depends on how full it is (but it will resize if it gets too full). Also, speed of your machine or your network connection or of a remote repo server might also fluctuate due to other loads.

Archives might be very different in their unique chunks amount and size, some archives might be very similar than the previous ones while others might have a lot of new/unique chunks.

borg2 will still take quite a while (depends a bit on whether some more bigger breaking changes will get into it or not).

**Reply by mbwgh** [reply]
Thanks for the swift reply.
I believe the slowdown I observed was due to the fact that I arrived at backups that were done during a time where I had to do a lot of screen recordings, so that would coincide with lots of new chunks.

Unfortunately, successive backups where this files remained on disk (and were backed up again) are still pretty slow, since processing these files and comparing against already existing chunks takes time. I guess the saving grace is knowing that I have not hit some exponential wall and will just need to slog through it.
